reference/interbit-ui-components/CONTRIBUTING.adoc

```diff
@@ -0,0 +1,205 @@
+`src/components`. Each component must be exported from `src/index.js`
+
+==== SCSS
+
+For each component:
+- The styles should be in their own file, say `_my-component.scss`, in the
+`src/scss/interbit` directory. This file should be imported in
+`src/scss/interbit.scss`.
+- The styles should be contained in a class named after the component and
+prefixed with `ibweb-`. For example, `MyComponent` should have a top-level
+class called `ibweb-my-component`. See `src/scss/interbit/_footer.scss` for how
+styles should be nested.
+- The CSS must be supported by the latest versions of Chrome, Firefox, Safari,
+and Edge. If in doubt, see link:https://caniuse.com/[caniuse.com].
+
+==== Tests
+
+Components must have an export test in `src/tests/exports.test.js` for each
+(sub)-component exported from `src/index.js`.
+
+UI Kit components must have, at the minimum, rendering tests in
+`src/tests/uiKit.test.js`. Interbit UI Components uses Jest and Enzyme for
+testing and rendering components.
+
+==== Documentation
+
+Interbit pulls documentation out of JSDoc style code comments. For each
+component:
+- Give a brief description of the component before the class declaration.
+- Give a description of each prop that the component accepts
+
+For example, the `LinkBar` component has the following code comments.
+```js
+/**
+ * A call-to-action UI component. It functions as an `<a>` tag or `<Link>` component from `react-router-dom`, together with some additional presentational information (title, description, and image).
+ */
+export default class LinkBar extends Component {
+  static propTypes = {
+    /**
+     * A short title describing the call-to-action for LinkBar
+     */
+    title: PropTypes.string.isRequired,
+    /**
+     * A description of the call-to-action for LinkBar
+     */
+    content: PropTypes.string.isRequired,
+    /**
+     * An image that provides context for the call-to-action
+     */
+    image: PropTypes.string,
+    /**
+     * A URL, relative path, or mailto link. E.g. `https://interbit.io`,  `/pages/foo`, or `mailto:foo@interbit.io`
+     */
+    to: PropTypes.string,
+    /**
+     * CSS classes to be applied to the component
+     */
+    className: PropTypes.string,
+    /**
+     * A flag. Set to `true` if the `to` property is a `mailto:` link
+     */
+    isMailto: PropTypes.bool,
+    /**
+     * A function that is fired when a user clicks on the LinkBar
+     */
+    clickHandler: PropTypes.func
+  }
+...
+}
+```
+
+Each UI Kit component should be accompanied by an example file,
+`MyComponent.md`. link:https://react-styleguidist.js.org/docs/documenting.html[React Styleguidist]
+will render the components and JSDoc comments in a style guide.
+
+The example file should include:
+- An example of the rendered component.
+- An example of each variant of the component. For example, we might apply
+`className="blue"` to a button component to render a blue button.
+
+The following is `LinkBar`'s example `md` file:
+
+````
```

Comment:
This doesn't render correctly in GitHub. You see a pair of backticks.

Reply:
Backticks are the "compatibility" markup for source code blocks. Ideally, we'd use Asciidoctor block demarcation:

```
[source,js]
----
Content containing backticks, source code, etc.
----
```

See: https://asciidoctor.org/docs/user-manual/#delimited-blocks